### PR TITLE
Reafactor GraphQL initializing

### PR DESCRIPTION
**Pull request by kirill-oleynik**
(Previous PR)

### CODE REVIEW

File: app/controllers/api/v1/api_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    case variables_param
+    when String
+      variables_param.present? ? variables_param(JSON.parse(variables_param)) : {}
+    when Hash, ActionController::Parameters
+      variables_param
+    else
+      raise ArgumentError, "Unexpected parameter: #{variables_param}"
+    end
+  end
+end
```
Comment (kolyanjy) [inline app/controllers/api/v1/api_controller.rb:26]: this must be in graphql controller, but graphql_controller must be inherit of api_controller

File: app/graphql/to_do_lists_internship_schema.rb [hunk 2]
```diff
@@ -2,35 +2,12 @@
 
 # Application GraphQL Schema
 class ToDoListsInternshipSchema < GraphQL::Schema
+  SHEMA_SETTINGS = {
+    MAX_COMPLEXITY: 100,
+    MAX_DEPTH: 10
+  }.freeze
```
Comment (kolyanjy) [inline app/graphql/to_do_lists_internship_schema.rb:8]: I meant separate file with constants, but it's ok to

File: config/initializers/constants.rb [hunk 3]
```diff
@@ -3,4 +3,10 @@ module User
     PASSWORD_MIN_SIZE = 6
     EMAIL_REGEXP = /.+@.+\..+/.freeze
   end
+  module GraphQL
+    SHEMA_SETTINGS = {
```
Comment (kolyanjy) [inline config/initializers/constants.rb:7]: typo: schema

File: app/controllers/graphql_controller.rb [hunk 4]
```diff
@@ -1,53 +1,26 @@
 # frozen_string_literal: true
 
-class GraphqlController < ApplicationController
-  # If accessing from outside this domain, nullify the session
-  # This allows for outside API access while preventing CSRF attacks,
-  # but you'll have to authenticate your user separately
-  # protect_from_forgery with: :null_session
-
+class APIController < APIController
```
Comment (kolyanjy) [inline app/controllers/graphql_controller.rb:3]: APIController < APIController ?

Comment (kirill-oleynik) [reply app/controllers/graphql_controller.rb:3]: I need to be more attentive and less in a hurry 🤣

File: app/controllers/api/v1/api_controller.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class APIController < ApplicationController
```
Comment (kolyanjy) [inline app/controllers/api/v1/api_controller.rb:5]: `ApiController` convention for zeitwerk

File: app/controllers/api/v1/api_controller.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+module API
```
Comment (kolyanjy) [inline app/controllers/api/v1/api_controller.rb:3]: Api

File: app/controllers/graphql_controller.rb [hunk 7]
```diff
@@ -1,53 +1,26 @@
 # frozen_string_literal: true
 
-class GraphqlController < ApplicationController
-  # If accessing from outside this domain, nullify the session
-  # This allows for outside API access while preventing CSRF attacks,
-  # but you'll have to authenticate your user separately
-  # protect_from_forgery with: :null_session
-
+class GraphQLController < ::API::V1::APIController
```
Comment (kolyanjy) [inline app/controllers/graphql_controller.rb:3]: Api::V1::ApiController

**Review by kolyanjy — APPROVED** [review]

**Pull request merged**